### DISCUSSION TITLE
Building CGNS Lib for CentOS - Build Error

### DISCUSSION THREAD

**Discussion by Peter-P779** [CGNS/CGNS#347] [CGNStalk]
Hello,

How to build the lib for CentOS?

I did:
- clone repos from github
- change into the repos folder
- cmake src
- ccmake src
- make

During the build i get the building error:

`_hashmap.c(167): error: expected an expression
  #if SIZEOF_LONG == SIZEOF_VOID_P
                  ^
`

**Comment by MicK7** [comment]
https://cgns.github.io/cgns-modern.github.io/general/FAQ.html#installation-of-cgns-from-source

You should do an out of tree build as explain in the doc.
